addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2484,7 +2496,19 @@ options.registry.CoverProperties = options.Class.extend({
      */
     updateUI: async function () {
         await this._super(...arguments);
+        this._updateSavingDataset();
+    },
 
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Updates the cover properties dataset used for saving.
+     *
+     * @private
+     */
+    _updateSavingDataset: function (colorFromPreview) {
```

Comment:
What this unused parameter?

Also, I don't why this fix would be necessary. Can you explain? Normally, updateUI must be called each time there is a non preview update

Reply:
The `custom_color_picked`'s handler explicitly mentions in its comment that it only updates the value when the palette popup is closed, on purpose. To achieve this, it does not call `_onUserValueChange()`.

Upon Save, the widgets are all destroyed, but this does not trigger the actual close effect on the palette popup. I will look into a way to trigger the close (if opened) during the destroy... but it sounds a bit strange having `destroy` that triggers `updateUI`. (After the `destroy`, the `cleanForSave` also closes stuff but: the event listeners have already been removed by the `destroy`, and there is a `setTimeout` on the way that defers the triggering until after the page is saved in reloaded)

Reply:
Mmh ok, I guess this is kinda like the we-input which trigger preview mode when typing text inside and if you hit save, it works because the input had to be unfocused and thus trigger the non-preview update first.

I agree updateUI should not have to be called for the save to make sense (this is a hack here), maybe a cleaner solution would then be to call your _updateSavingDataset everywhere it make sense and not in updateUI... to check :+1:

Reply:
After trying to `_updateSavingDataset` after the general `_select` (through a `.then(...)`), I realise this does not work for the Size and Filter intensity options because their new state is only available after `updateUI` was called, but it is called behind a `setTimeout`. (It however works for the background color because the internal state of the color picker is accessed directly.)
For it to work, `_updateSavingDataset` would have to be based on the targeted elements instead of the options. Would that make sense ?
(Otherwise, I would surprisingly end up in the state of the current commit - maybe with additional comments to better explain what is going on.)

Reply:
> For it to work, _updateSavingDataset would have to be based on the targeted elements instead of the options. 

If you mean according to the DOM of the website, yes, this is probably the best solution? Any downside?